sqlmesh/core/plan/definition.py

```diff
@@ -332,7 +332,10 @@ def has_changes(self) -> bool:
 
     def is_new_snapshot(self, snapshot: Snapshot) -> bool:
         """Returns True if the given snapshot is a new snapshot in this plan."""
-        return snapshot.snapshot_id in {s.snapshot_id for s in self.new_snapshots}
```

Comment:
@eakmanrq you missed this section in your refactor, it's an N^2 operation

Reply:
Thanks for catching. Want to just update `new_snapshots` so it is a dict of snapshot ID to snapshot?

Reply:
ok i can do that

Reply:
decided not to do that because then i have to change other places to values